No support for 'preview' language versions?

I'm writing some code to try out the new 'unmanaged constructed types' and 'static local functions' features, and as expected OmniSharp is flagging this up with error CS8652, stating that the feature is not supported and currently in preview.

However, the error says to use the `preview` language version to use these features, and upon specifying this in the `LangVersion` property, these errors remain. To hide these errors, I had to specify `8.0` as the value, instead of what the error had specified.

I believe this is due to this switch not including `preview` as a possible value, though I may be wrong since I'm not familiar with the code:
https://github.com/OmniSharp/omnisharp-roslyn/blob/46c814a74c0dae8f4c2541bff7c65f686597af04/src/OmniSharp.MSBuild/ProjectFile/PropertyConverter.cs#L38-L53

--- a/src/OmniSharp.MSBuild/ProjectFile/PropertyConverter.cs
+++ b/src/OmniSharp.MSBuild/ProjectFile/PropertyConverter.cs
@@ -28,29 +28,10 @@ namespace OmniSharp.MSBuild.ProjectFile
 
         public static LanguageVersion ToLanguageVersion(string propertyValue)
         {
-            if (string.IsNullOrWhiteSpace(propertyValue) ||
-                propertyValue.Equals("Default", StringComparison.OrdinalIgnoreCase))
-            {
-                return LanguageVersion.Default;
-            }
+            if (LanguageVersionFacts.TryParse(propertyValue, out var result))
+                return result;
 
-            // ISO-1, ISO-2, 3, 4, 5, 6, 7 or Default
-            switch (propertyValue.ToLower())
-            {
-                case "latest": return LanguageVersion.Latest;
-                case "iso-1": return LanguageVersion.CSharp1;
-                case "iso-2": return LanguageVersion.CSharp2;
-                case "3": return LanguageVersion.CSharp3;
-                case "4": return LanguageVersion.CSharp4;
-                case "5": return LanguageVersion.CSharp5;
-                case "6": return LanguageVersion.CSharp6;
-                case "7": return LanguageVersion.CSharp7;
-                case "7.1": return LanguageVersion.CSharp7_1;
-                case "7.2": return LanguageVersion.CSharp7_2;
-                case "7.3": return LanguageVersion.CSharp7_3;
-                case "8.0": return LanguageVersion.CSharp8;
-                default: return LanguageVersion.Default;
-            }
+            return LanguageVersion.Default;
         }
 
         public static ImmutableArray<string> SplitList(string propertyValue, char separator)

